sklearn/utils/_fast_dict.pyx

```diff
@@ -91,10 +87,8 @@ cdef class IntFloatDict:
                 The values of the data points
         """
         cdef int size = self.my_map.size()
-        cdef cnp.ndarray[ITYPE_t, ndim=1] keys = np.empty(size,
-                                                         dtype=np.intp)
-        cdef cnp.ndarray[DTYPE_t, ndim=1] values = np.empty(size,
-                                                           dtype=np.float64)
+        cdef ITYPE_t[:] keys = np.empty(size, dtype=np.intp)
+        cdef DTYPE_t[:] values = np.empty(size, dtype=np.float64)
```

Comment:
As is, this function would return memviews and not ndarrays.
```suggestion
        keys = np.empty(size, dtype=np.intp)
        values = np.empty(size, dtype=np.float64)
```

However, this method is not tested and not used anywhere. I wonder if we should just remove it.

Reply:
I think if it is not used anywhere it would make sense to remove it.

Reply:
I have removed the method but I was wondering that could there be a possible use case in which it can be used?

Reply:
Looking at this again, I think `to_arrays` was made to just be a wrapper around ``_to_arrays`` for debugging purpose. I think we can keep it and instead add a small test in ``test_fast_dict.py``. Something like:
```py
def test_to_arrays():
    keys_in = np.array([1, 2, 3], dtype=np.intp)
    values_in = np.array([4, 5, 6], dtype=np.float64)

    d = IntFloatDict(keys_in, values_in)
    keys_out, values_out = d.to_arrays()

    assert_array_equal(keys_out, keys_in)
    assert_allclose(values_out, values_in)
```

Reply:
Right, agreed. Thank you for the feedback.